packages/wpcom-proxy-request/src/index.js

```diff
@@ -15,7 +15,7 @@ const proxyOrigin = 'https://public-api.wordpress.com';
 /**
  * "Origin" of the current HTML page.
  */
-const origin = window.location.protocol + '//' + window.location.host;
+const origin = globalThis?.location?.protocol + '//' + globalThis?.location?.host;
```

Comment:
@jsnajdr, I'd love your (or someone else's from Team Calypso) expert opinion on this. If I don't make this change, it will trigger an "Undefined  error while trying to render SSR, it seems:

```
/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/packages/wpcom-proxy-request/src/index.js:18
const origin = window?.location?.protocol + '//' + window?.location?.host;
               ^
ReferenceError: window is not defined
    at Object../client/components/inline-support-link/index.jsx (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/packages/wpcom-proxy-request/src/index.js:18:16)
    at __webpack_require__ (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/webpack/bootstrap:19:1)
    at Object../client/my-sites/plugins/index.node.js (/Users/klimeryk/work/workspace/calypso/build/server.js:45587:27)
    at __webpack_require__ (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/webpack/bootstrap:19:1)
    at Object.load (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/client/sections.js:146:17)
    at forEach (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/client/server/pages/index.js:908:13)
    at Array.forEach (<anonymous>)
    at pages (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/client/server/pages/index.js:899:4)
    at boot (/Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/client/server/boot/index.js:96:11)
    at /Users/klimeryk/work/workspace/calypso/build/webpack:/wp-calypso/client/server/index.js:14:13
```

Looks like this line is what ultimately leads to loading of wpcom-proxy-request in this context:

https://github.com/Automattic/wp-calypso/pull/79741/files#diff-8b10c9eab3990cb2c7310ada64772c069aafd016f76fc7692328b5f148cd6a98R2

I found this related Calypso PR https://github.com/Automattic/wp-calypso/pull/65576, which led to a suggestion to use `globalThis` from you (p4TIVU-a9o-p2#comment-11513).

Reply:
This solution is fine for sure, but it's a bit of a bandage.

`origin` is only used twice, both don't need to be in module initialization. I'd inline `origin` to where it is used, and then it can stay `window`. 

Maybe move `debug( 'using "origin": %o', origin )` inside `install` and use `window.location.protocol + '//' + window.location.host` right away.

Reply:
Thanks for the nudge - I thought about it initially, but wanted to keep the changes to minimum and avoid changes to behaviour. I was concerned that there's some edge cases where the value at the time of module initialization will differ from the one inline. But it makes more sense and is clearer, so let's try it 💪 

I've extracted these changes to a separate PR: https://github.com/Automattic/wp-calypso/pull/79789. That way it's easier to test, approve and revert if needed.